src/core/thread/mlr_manager.cpp

```diff
@@ -475,6 +477,8 @@ void MlrManager::HandleMulticastListenerRegistrationResponse(Coap::Message *
     Ip6::Address failedAddresses[kIp6AddressesNumMax];
     uint8_t      failedAddressNum = 0;
 
+    failedAddresses[0] = {}; // suppress -Wuninitialized
```

Comment:
How about using `failedAddresses[0].Clear()` instead?

(while `failedAddresses[0] = {};` works fine, some compiler can generate very inefficient code for this, so as pattern I recommend not using it. They basically create a new `Ip6::Address` instance init it with `{}` and then call copy constructor to copy it to `failedAddresses[0]`).

Reply:
Good idea!